polaris-service/src/test/java/org/apache/polaris/service/task/ManifestFileCleanupTaskHandlerTest.java

```diff
@@ -226,4 +233,228 @@ public void deleteFile(String location) {
       assertThatPredicate((String f) -> TaskUtils.exists(f, fileIO)).rejects(dataFile2Path);
     }
   }
+
```

Comment:
The test cases in this file primarily verify theses cleanup:
1. straightforward cleanup
2. cleanup when files do not exist
3. cleanup with retries